pallets/committee/src/lib.rs

```diff
@@ -216,9 +230,13 @@ pub mod pallet {
         /// the committee
         pub fn ensure_member(
             origin: OriginFor<T>,
+            council: bool,
         ) -> Result<CommitteeMember<AccountIdFor<T>>, DispatchError> {
             let who = ensure_signed(origin)?;
             if let Some(member_type) = Members::<T>::get(who.clone()) {
+                let _ = council
+                    && member_type == MemberType::Constituent
+                    && Err(<Error<T>>::NotCouncilMember)?;
```

Comment:
This is rather hard to read. Should me make two designated functions for `ensure_council` and `ensure_consituent`instead?

Reply:
I like the current implementation as I provided, but I'm okay with changing it to `ensure_council` and `ensure_consituent`